lib/queue.rb

```diff
@@ -2,30 +2,65 @@ class Queue
 
   def initialize
     # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    # raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1
+    # @front = -1
+    # @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+      @store[@front] = element
+    elsif @front == @back
+      #DECIDE
+      raise ArgumentError, "Queue is full!"
+    else
+      @store[@back] = element
+      @back = (@back + 1) % @store.length
+    end
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    # raise NotImplementedError, "Not yet implemented"
+    if @front == @back
+      raise ArgumentError, "Queue is empty!"
+    end
+    if element = @store[@front]
+      @front = (@front + 1) % @store.length
+      return element
```

Comment:
The problem here is determining if the queue is empty or full because in either case front would be equal to back.

So

```suggestion
    if element = @store[@front]
      @front = (@front + 1) % @store.length
      @front = @back = -1 if @front == @back
      return element
```